File: Java/Bitwise/Binary Number with Alternating Bits.cpp
class Solution {
    String helper(int n){
        String res="";
        while(n!=0){
            int bit=n&1;
            res=Integer.toString(bit)+res;
            n=n>>1;
        }
        return res;
    }
    public boolean hasAlternatingBits(int n) {
        String str=helper(n);
        for(int i=0;i<str.length()-1;i++){
            if(str.charAt(i)==str.charAt(i+1)){
                return false;
            }
        }
        return true;
    }
}

File: Python/Bitwise/Binary Number with Alternating Bits.py
class Solution:
    def helper(self,n:int)->str:
        res=""
        while n!=0:
            bit=n&1
            res=str(bit)+res
            n=n>>1
        return res
    def hasAlternatingBits(self, n: int) -> bool:
        str=self.helper(n)
        for i in range(len(str)-1):
            if str[i]==str[i+1]:
                return False
        return True
        
